### ISSUE TITLE
lint: add underef checker

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#14]
Find dereference expressions  (`ast.StarExpr`) that can be simplified.

All these cases must be detected:
```go
/*1*/ (*x)[i] => x[i] // If `x` is an array
/*2*/ (*p).f  => p.f  // If `p` is a pointer to type with `f` field
```
There could be more.

**Comment by fexolm** [comment]
(*p).f fixed in #42
